internal/resources/cluster/resource_cluster.go

```diff
@@ -346,12 +358,25 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, m interf
 			Summary:  "Kubernetes cluster's kubeconfig provided. Proceeding to attach the cluster TMC",
 		})
 
-		deploymentManifests, err := manifest.GetK8sManifest(clusterResponse.Cluster.Status.InstallerLink)
+		if clusterResponse.Cluster.Spec.ImageRegistry != "" {
+			clusterManifest, err := config.TMCConnection.ManifestResourceService.ClusterManifestHelperGetManifest(constructFullname(d))
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "Unable to get manifest (%s) for cluster entry, name : %s", clusterManifest.Manifest, err))
+			}
+
+			manifests = clusterManifest.Manifest
+		} else {
+			manifests = clusterResponse.Cluster.Status.InstallerLink
+		}
+
+		log.Printf("[INFO] Manifest lir-manifest is %s ", manifests)
+
+		deploymentManifests, err := manifest.GetK8sManifest(manifests)
 		if err != nil {
 			return append(diags, diag.FromErr(err)...)
 		}
 
-		log.Printf("[INFO] Applying %s cluster's deployment link manifest objects on to kubernetes cluster", constructFullname(d).ToString())
+		log.Printf("[INFO] Applying %s cluster's deployment link manifest objects on to kubernetes cluster and manifest is %s", constructFullname(d).ToString(), manifests)
```

Comment:
I believe again here you are logging the manifest for debugging purpose?

Reply:
Yes, removed